src/Compilers/CSharp/Portable/BoundTree/BoundNodes.xml

```diff
@@ -1830,6 +1830,30 @@
     <Field Name="WasTargetTyped" Type="bool" PropertyOverrides="true"/>
   </Node>
   
+  <!--
+  A target-typed collection literal expression.
+  It does not survive past initial binding.
+  -->
+  <Node Name="BoundUnconvertedCollectionLiteralExpression" Base="BoundExpression">
+    <!-- Type is not significant for this node type; always null -->
+    <Field Name="Type" Type="TypeSymbol?" Override="true" Null="always"/>
+    <!-- Collection literal elements. -->
+    <Field Name="Initializers" Type="ImmutableArray&lt;BoundExpression&gt;"/>
+    <Field Name="Binder" Type="Binder" Null="disallow"/>
+  </Node>
+
+  <Node Name="BoundCollectionLiteralExpression" Base="BoundExpression">
+    <!-- Non-null type is required for this node kind -->
+    <Field Name="Type" Type="TypeSymbol" Override="true" Null="disallow"/>
+    <!-- CollectionCreation will be null for arrays that are constructed from initializers directly and for type parameters. -->
+    <Field Name="CollectionCreation" Type="BoundExpression?" Null="allow" SkipInVisitor="true"/>
+    <!-- Constructor to create a span from an array. -->
+    <Field Name="SpanConstructor" Type="MethodSymbol?" Null="allow"/>
+    <!-- An expression placeholder value representing the initialized collection. -->
+    <Field Name="Placeholder" Type="BoundObjectOrCollectionValuePlaceholder" Null="disallow" SkipInVisitor="true"/>
+    <!-- Collection literal elements. -->
+    <Field Name="Initializers" Type="ImmutableArray&lt;BoundExpression&gt;"/>
+  </Node>
```

Comment:
to me, it makes sense to perhaps have an abstract base type for the BoundCollectionLiteralExpression and a few specific concrete subclasses for the different cases (Array/span, vs collection-initializer).  That way you don't have to know which properties are value for which types of collection literals.